src/test/unit_tests/test_chunk_fs.js

```diff
@@ -19,4 +19,12 @@ mocha.describe('ChunkFS', function() {
         self.timeout(RUN_TIMEOUT);
         await chunk_fs_hashing.file_target();
     });
+
+    mocha.it('Concurrent ChunkFS with file target - produce more than 1024 chunks', async function() {
+        const self = this;
+        self.timeout(RUN_TIMEOUT);
+        const chunk_size = 100;
+        const parts_s = 1;
+        await chunk_fs_hashing.file_target(chunk_size, parts_s);
+    });
```

Comment:
I might be missing something, but how is this causing more than 1024 chunks?

Reply:
so it's a bit complex I'll explain:
My goal was to implement a test that will cause evalutation of ```this.q_buffers.length === iov_max``` to true before ```this.q_size >= config.NSFS_BUF_SIZE``` evaluates to true.
So what I had to do is to create more than 1024 chunks of small sizes.
Using the default values:
num of chunkes = (20*1024*1024)/(16*1024) = 1280
chunk size = 16*1024
NSFS_BUF_SIZE = 8*1024*1024
which means that using the default values after 512 chunks ```this.q_size >= config.NSFS_BUF_SIZE``` will evaluate to true. So I had to reduce chunk size to 100.
num of chunks = (10*1024*1024)/100 = 104857
chunk size = 100
which means that using the new value ```this.q_buffers.length === iov_max``` will evaluate after 1024 chunks and the size will be 1024*100 and ```this.q_size >= config.NSFS_BUF_SIZE``` is still false

Reply:
I see - worth copying this explanation to a code comment.

Reply:
added